### PR TITLE
Update core API in bindings core

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Update core API in bindings core to https://editor.swagger.io/?url=https://raw.githubusercontent.com/iotaledger/tips/tip48/tips/TIP-0048/openapi3-core.yaml

## Links to any relevant issues

Required for #1696 

## Type of change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: bindings/core/src/method_handler/client.rs [hunk 1]
```diff
@@ -173,7 +173,21 @@ pub(crate) async fn call_client_method_internal(client: &Client, method: ClientM
         ClientMethod::GetHealth { url } => Response::Bool(client.get_health(&url).await?),
         ClientMethod::GetNodeInfo { url, auth } => Response::NodeInfo(Client::get_node_info(&url, auth).await?),
         ClientMethod::GetInfo => Response::Info(client.get_info().await?),
-        ClientMethod::GetPeers => Response::Peers(client.get_peers().await?),
+        ClientMethod::GetAccountCongestion { address } => {
+            // TODO: .as_account() can panic, accept account id or bech32 address?
```
Comment (Thoralf-M) [inline bindings/core/src/method_handler/client.rs]: @thibault-martinez would you also accept the account id from the bindings side? Or should I just make the conversion return an error if a wrong address is provided?

File: sdk/src/client/node_api/core/routes.rs [hunk 2]
```diff
@@ -293,6 +293,17 @@ impl ClientInner {
         self.get_request(path, None, true, true).await
     }
 
+    /// Find the metadata of the earliest block containing the transaction that was confirmed.
```
Comment (thibault-martinez) [inline sdk/src/client/node_api/core/routes.rs]: block?

**Review by thibault-martinez — COMMENTED** [review]
There is some peer related stuff in the nodejs bindings if you want to remove it as well

File: sdk/src/client/node_api/core/routes.rs (hunk 2 above)
Comment (Thoralf-M) [reply sdk/src/client/node_api/core/routes.rs]: That's wrong, already commented on the TIP, but not sure what exactly it should be

**Comment by Thoralf-M** [comment]
> There is some peer related stuff in the nodejs bindings if you want to remove it as well

Would do that in another PR for nodejs alone when I add the new routes there

File: bindings/core/src/method_handler/client.rs (hunk 1 above)
Comment (thibault-martinez) [reply bindings/core/src/method_handler/client.rs]: I would personally take IDs

File: sdk/src/client/node_api/core/routes.rs (hunk 2 above)
Comment (Thoralf-M) [reply sdk/src/client/node_api/core/routes.rs]: Updated to something that makes sense

File: sdk/src/client/node_api/core/routes.rs [hunk 3]
```diff
@@ -293,6 +293,17 @@ impl ClientInner {
         self.get_request(path, None, true, true).await
     }
 
+    /// Find the metadata of a transaction.
```
Comment (thibault-martinez) [inline sdk/src/client/node_api/core/routes.rs]:
```suggestion
    /// Finds the metadata of a transaction.
```

File: bindings/core/src/response.rs [hunk 4]
```diff
@@ -101,8 +102,20 @@ pub enum Response {
     /// - [`GetInfo`](crate::method::ClientMethod::GetInfo)
     Info(NodeInfoWrapper),
     /// Response for:
-    /// - [`GetPeers`](crate::method::ClientMethod::GetPeers)
-    Peers(Vec<PeerResponse>),
+    /// - [`GetAccountCongestion`](crate::method::ClientMethod::GetAccountCongestion)
+    CongestionResponse(CongestionResponse),
```
Comment (thibault-martinez) [inline bindings/core/src/response.rs]: Remove response suffix

File: bindings/core/src/method/client.rs [hunk 5]
```diff
@@ -148,8 +149,45 @@ pub enum ClientMethod {
...
+    /// Returns information of all registered validators and if they are active, ordered by their holding stake.
+    #[serde(rename_all = "camelCase")]
+    GetValidators {
+        /// The page number of validators.
+        page_size: Option<u32>,
+        /// Starts the search from the cursor (requested slot index+start index).
+        cursor: Option<String>,
+    },
+    #[serde(rename_all = "camelCase")]
+    /// Return information about a validator.
```
Comment (thibault-martinez) [inline bindings/core/src/method/client.rs:178]: Comment before

File: bindings/core/src/method_handler/client.rs [hunk 6]
```diff
@@ -208,12 +218,30 @@ pub(crate) async fn call_client_method_internal(client: &Client, method: ClientM
...
         ClientMethod::GetIncludedBlock { transaction_id } => {
             Response::Block(BlockDto::from(&client.get_included_block(&transaction_id).await?))
         }
         ClientMethod::GetIncludedBlockMetadata { transaction_id } => {
             Response::BlockMetadata(client.get_included_block_metadata(&transaction_id).await?)
         }
+        ClientMethod::GetTransactionMetadata { transaction_id } => {
+            Response::TransactionMetadata(client.get_transaction_metadata(&transaction_id).await?)
+        }
+        ClientMethod::GetCommitments { commitment_id } => {
```
Comment (thibault-martinez) [inline bindings/core/src/method_handler/client.rs]: GetCommitment?

Comment (Thoralf-M) [reply bindings/core/src/method_handler/client.rs]: Yes

**Review by thibault-martinez — APPROVED** [review]

**Review by kwek20 — COMMENTED** [review]
i dont get it, surely there should be an api call for peers and metric? Are we sure its gone and not just forgotten? It definitely still exists in the server code :)

**Comment by Thoralf-M** [comment]
@kwek20 they got removed from the core API and there is an own management API, but I don't think we need anything of that

**Review by kwek20 — APPROVED** [review]

**Pull request merged**